tests/tt_metal/tt_metal/api/test_dram.cpp

```
@@ -287,27 +288,31 @@ TEST_F(MeshDispatchFixture, TensixDRAMLoopbackSingleCoreDB) {
 TEST_F(MeshDispatchFixture, ActiveEthDRAMLoopbackSingleCore) {
     constexpr uint32_t buffer_size = 2 * 1024 * 25;
 
-    if (!this->IsSlowDispatch()) {
-        log_info(tt::LogTest, "This test is only supported in slow dispatch mode");
-        GTEST_SKIP();
-    }
-
     unit_tests_common::dram::test_dram::DRAMConfig dram_test_config = {
         .core_range = {{0, 0}, {0, 0}},
         .kernel_file = "tests/tt_metal/tt_metal/test_kernels/dataflow/dram_copy.cpp",
         .dram_buffer_size = buffer_size,
         .l1_buffer_addr = tt::align(
             tt::tt_metal::hal::get_erisc_l1_unreserved_base(),
             MetalContext::instance().hal().get_alignment(HalMemType::DRAM)),
-        .kernel_cfg = tt_metal::EthernetConfig{.eth_mode = Eth::RECEIVER, .noc = tt_metal::NOC::NOC_0},
+        .kernel_cfg = tt_metal::EthernetConfig{},
     };
 
     for (auto mesh_device : devices_) {
         auto device = mesh_device->get_devices()[0];
         for (auto active_eth_core : device->get_active_ethernet_cores(true)) {
             log_info(tt::LogTest, "Active Eth Loopback. Logical core {}", active_eth_core.str());
             dram_test_config.core_range = {active_eth_core, active_eth_core};
-            ASSERT_TRUE(unit_tests_common::dram::test_dram::dram_single_core(this, mesh_device, dram_test_config));
+            const auto erisc_count = tt::tt_metal::MetalContext::instance().hal().get_processor_classes_count(
```

nitpick: can you use the hal function that returns the total number of riscvs instead of number of processor classes?  I think the number of HAL processor classes will eventually be fixed to be 1 on eth cores (only DM), which will then silently skip erisc1 here.